compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -903,15 +905,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             // We are inside a function body, so reporting "return statement
             // outside of function body" needs an explanation.
 
-            let encl_body_owner_id = self.tcx.hir().enclosing_body_owner(expr.hir_id);
-
             // If this didn't hold, we would not have to report an error in
             // the first place.
-            assert_ne!(encl_item_id.def_id, encl_body_owner_id);
+            assert_ne!(encl_item_id.def_id, self.body_id);
 
-            let encl_body = self.tcx.hir().body_owned_by(encl_body_owner_id);
-
-            err.encl_body_span = Some(encl_body.value.span);
+            err.encl_body_span = Some(self.tcx.def_span(self.body_id));
```

Comment:
This change is because `enclosing_body_owner` does not yield const blocks anymore. I'm addressing this separately to remove this footgun, but the code here could have been written simpler anyway, so I just did that.

Reply:
`enclosing_body_owner` being out of sync is somewhat sketch 💀 why doesn't it? it should be as much of a body as a closure is, for example.